### PR TITLE
Update migration to use the web template as a base for web projects

### LINKED ISSUE #4492: Open Dev14 ASP.NET Core web app using Dev15 RC, dotnet migrate should add the PackageReference to Microsoft.NET.Sdk.Web
**INSTALL STEPS**
1. Run preinstall.cmd
2. Install Dev15 RC d15rel.25822.01 (.NET Core workload)
3. Setup private Myget feeds

**REPRO STEPS**
1. Copy Dev14 ASP.NET Core solution to a local folder (see bug 281338 in TFS for the repro project or contact me)
    - Use AspNetCoreWebApp_NoAuth_1-- it is the default Web App (no auth) template created using Dev14 K101 VS Tooling
    - You can use other solutions in this share too for additional testing
2. Open solution

**EXPECTED**
Package reference to Microsoft.NET.Sdk.Web to get added

**ACTUAL**
Package reference to Microsoft.NETCore.Sdk gets added

### PR DESCRIPTION
**Pull request by mlorbetske**
Quite probably not ready to merge yet, putting it out there to get feedback

Detects web projects by:
* Checking for an xproj with the web targets (definitive)
* Checking for any of the following strings in a project.json (less definitive, also checks for the string `netcoreapp`)
  * wwwroot
  * BrowserLink
  * BundlerMinifier
  * AspNetCore

Fixes #4492

/cc: @livarcocc @piotrpMSFT @BillHiebert

### CODE REVIEW

**Comment by livarcocc** [comment]
Can you add some tests for this? Have you tried migrating web apps with your changes?

**Comment by mlorbetske** [comment]
cc @prafullbhosale

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectTypeDetector.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        private bool IsWebProject(string xprojFile, string projectJsonFile)
+        {
+            if (xprojFile != null && File.Exists(xprojFile))
+            {
+                var xprojText = File.ReadAllText(xprojFile);
+                return xprojText.IndexOf(WebTargets, StringComparison.OrdinalIgnoreCase) > -1;
+            }
+
+            string projectJsonText = File.ReadAllText(projectJsonFile);
+            return projectJsonText.IndexOf("netcoreapp", StringComparison.OrdinalIgnoreCase) > -1
```
Comment (eerhardt) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectTypeDetector.cs]: Does this work for desktop web projects?

**Comment by livarcocc** [comment]
@dotnet-bot Test CentOS7.1 x64 Debug Build
@dotnet-bot Test Ubuntu16.04 x64 Debug Build

**Pull request closed without merging**